lib/fencing/st_client.c

```diff
@@ -1749,10 +1749,13 @@ stonith__validate(stonith_t *st, int call_options, const char *rsc_id,
      * that is incorrect, we will need to allow the caller to pass the target).
      */
     const char *target = "node1";
+    const char *host_arg = NULL;
 
     if (params != NULL) {
         /* Remove special stonith params from the table before doing anything else */
         g_hash_table_foreach_remove(params, is_stonith_param, NULL);
+
+        host_arg = g_hash_table_lookup(params, PCMK_STONITH_HOST_ARGUMENT);
```

Comment:
that's a stonith param, you'll need to look it up before removing it :) (and make a copy)